source/base/function_signed_distance.cc

```diff
@@ -332,6 +332,88 @@ namespace Functions
 
       return evaluate_ellipsoid(point) < 0.0 ? -distance : distance;
     }
+
+
+
+    template <int dim>
+    Rectangle<dim>::Rectangle(const Point<dim> &bottom_left,
+                              const Point<dim> &top_right)
+      : bounding_box({bottom_left, top_right})
+    {}
+
+
+
+    template <int dim>
+    double
+    Rectangle<dim>::value(const Point<dim> & p,
+                          const unsigned int component) const
+    {
+      (void)component;
+
+      // calculate signed distance vector to bounding box
+      std::array<double, dim> distances;
+      for (unsigned int d = 0; d < dim; ++d)
+        distances[d] = bounding_box.signed_distance(p, d);
+
+      if (dim == 1)
+        {
+          // case: specialization for 1D
+          return distances[0];
+        }
+
+      unsigned int counter = 0;
+
+      for (unsigned int d = 0; d < dim; ++d)
+        if (distances[d] > 0)
+          counter++;
+
+      if (counter == 0)
+        {
+          // case: inside + on rectangle
+          double temp = std::abs(distances[0]);
+
+          for (unsigned int d = 1; d < dim; ++d)
+            temp = std::min(temp, std::abs(distances[d]));
```

Comment:
You know here that all of the distances are negative or zero. These lines are then equivalent to
```{suggestion}
  // All directional distances are zero or negative. The distance to the boundary
  // is the largest of these values.
  return *std::max_element (distances.begin(), distances.end());
```